litestar/typing.py

```diff
@@ -392,8 +393,8 @@ def is_non_string_collection(self) -> bool:
     def bound_types(self) -> tuple[FieldDefinition, ...] | None:
         """A tuple of bound types - if the annotation is a TypeVar with bound types, otherwise None."""
         if self.is_type_var and (bound := getattr(self.annotation, "__bound__", None)):
-            if is_non_string_sequence(bound):
-                return tuple(FieldDefinition.from_annotation(t) for t in bound)
+            if is_union(bound):
```

Comment:
This was a bug?

Reply:
Ah, yeah. I was trying to test this but found it didn't even work so I added this small fix